control-plane/pkg/reconciler/channel/channel.go

```diff
@@ -670,6 +714,38 @@ func (r *Reconciler) getSubscriptionName(channel *messagingv1beta1.KafkaChannel,
 	return "", apierrors.NewNotFound(messaging.SchemeGroupVersion.WithResource("subscriptions").GroupResource(), string(subscriber.UID))
 }
 
+func (r *Reconciler) getCaCerts() (string, error) {
+	secret, err := r.SecretLister.Secrets(system.Namespace()).Get(kafkaChannelTLSSecretName)
+	if err != nil {
+		return "", fmt.Errorf("failed to get CA certs from %s/%s: %w", system.Namespace(), kafkaChannelTLSSecretName, err)
+	}
+	caCerts, ok := secret.Data[caCertsSecretKey]
+	if !ok {
+		return "", fmt.Errorf("failed to get CA certs from %s/%s: missing %s key", system.Namespace(), kafkaChannelTLSSecretName, caCertsSecretKey)
+	}
+	return string(caCerts), nil
+}
+
+func (r *Reconciler) httpAddress(channel *corev1.Service) duckv1.Addressable {
+	// http address uses host-based routing
+	httpAddress := duckv1.Addressable{
+		Name: pointer.String("http"),
+		URL:  apis.HTTP(network.GetServiceHostname(channel.Name, channel.Namespace)),
+	}
+	return httpAddress
+}
+
+func (r *Reconciler) httpsAddress(caCerts string, channel *corev1.Service) duckv1.Addressable {
+	// https address uses path-based routing
+	httpsAddress := duckv1.Addressable{
+		Name:    pointer.String("https"),
+		URL:     apis.HTTPS(network.GetServiceHostname(NewChannelIngressServiceName, r.SystemNamespace)),
+		CACerts: pointer.String(caCerts),
+	}
+	httpsAddress.URL.Path = fmt.Sprintf("/%s/%s", channel.Namespace, channel.Name)
+	return httpsAddress
+}
```

Comment:
Can we replace these functions with the functions exported here? https://github.com/knative-sandbox/eventing-kafka-broker/blob/4f28710935edee6d9c4d7586ec18a3a2ccc38eec/control-plane/pkg/receiver/address.go#L38-L55